File: DFS.js
//DFS
var floodFill = function(image, sr, sc, color) {
    let originalColor = image[sr][sc]
    let newColor = color
    if(originalColor===newColor){
        return image
    }
    DFS(image, sr, sc, originalColor, newColor)
    return image
};

function DFS(image, r, c, originalColor, newColor){
    if(r<0||r>=image.length||c<0||c>=image[r].length){
        return
    }
    if(image[r][c]!==originalColor){
        return
    }
    image[r][c]=newColor
    DFS(image, r+1, c, originalColor, newColor)
    DFS(image, r, c+1, originalColor, newColor)
    DFS(image, r-1, c, originalColor, newColor)
    DFS(image, r, c-1, originalColor, newColor)
}

File: python3/algo/dfs.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        original = image[sr][sc]
        if original == color:
            return image
        def dfs(r:int, c:int):
            if r<0 or r>=len(image) or c<0 or c>=len(image[0]) or image[r][c]!=original:
                return
            image[r][c]=color
            dfs(r+1, c)
            dfs(r-1, c)
            dfs(r,c+1)
            dfs(r, c-1)
        dfs(sr, sc)
        return image